crates/iota-crypto/src/ed25519.rs

```diff
@@ -132,6 +134,30 @@ impl crate::PrivateKeyExt for Ed25519PrivateKey {
     }
 }
 
+impl crate::ConstPrivateKeyScheme for Ed25519PrivateKey {
+    const SCHEME: SignatureScheme = SignatureScheme::Ed25519;
+}
+
+#[cfg(feature = "mnemonic")]
+impl crate::FromMnemonic for Ed25519PrivateKey {
+    type Error = crate::PrivateKeyError;
+
+    fn from_mnemonic(phrase: &str) -> Result<Self, Self::Error> {
+        let mnemonic = bip32::Mnemonic::new(phrase, bip32::Language::English)?;
+        let seed = mnemonic.to_seed("");
+        Ok(Self::new(slip10_ed25519::derive_ed25519_private_key(
+            seed.as_bytes(),
+            &[
+                crate::DERIVATION_PATH_PURPOSE_ED25519,
+                crate::DERIVATION_PATH_COIN_TYPE,
+                0,
+                0,
+                0,
+            ],
```

Comment:
Fine to have this as default, but we should also allow other paths

Reply:
Hm not sure about this, the wallets we have only increase the account index for new accounts, maybe we should only allow this then so it's compatible
People that want something else can use a bip39 library directly to derive a private key with other bip32 paths

Reply:
Is this question still relevant?
> but we should also allow other paths

I see we are taking a path as parameter?

Reply:
Now we have both :)